src/cancelq/cancellable.h

```diff
@@ -0,0 +1,78 @@
+#ifndef CANCELLABLE_H
+#define CANCELLABLE_H
+
+#include <stddef.h>
+#include <stdbool.h>
+
+#define NOT_CANCELLABLE ((cancellable_id_t)-1)
+#define SCHEDULE_FAIL ((cancellable_id_t)-2)
+
+// The event queue that holds all these events
+typedef struct event_queue_s event_queue_t;
+
+// ID used to cancel cancellable_t
+typedef int cancellable_id_t;
+
+// Return a new event queue that will schedule and deliver events with
+// the eq functions below.
+event_queue_t *eq_new(void);
```

Comment:
I can't see `eq` without thinking "equal" or "equate".

Reply:
This isn't really a response to what you said here, but I think it's worth pointing out that `src/cancelq/cancellable.h` isn't a public interface. None of these symbols are even exported by `libsmear.a`.